### PR TITLE
implement real() and imag()

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

**Review by kaushikcfd — APPROVED** [review]
LGTM, thanks!

File: pytato/array.py [hunk 1]
```diff
@@ -569,6 +572,16 @@ def __abs__(self) -> Array:
     def __pos__(self) -> Array:
         return self
 
+    @property
+    def real(self) -> ArrayOrScalar:
```
Comment (kaushikcfd) [inline pytato/array.py:576]: Could you please add these to the docs.

Comment (matthiasdiener) [reply pytato/array.py:576]: Isn't that done here? https://github.com/inducer/pytato/pull/89/files#diff-edba40fd7883253a818c8fb49fabe814d360c14b89a23c82a47874b983170c36R360

Comment (kaushikcfd) [reply pytato/array.py:576]: Aah I meant the `pt.(real|imag)`.

Comment (matthiasdiener) [reply pytato/array.py:578]: oh, sure, fixed.

File: pytato/array.py [hunk 2]
```diff
@@ -1926,6 +1941,22 @@ def log10(x: Array) -> ArrayOrScalar:
...
+    else:
+        result_dtype = x.dtype
+    return _apply_elem_wise_func((x,), "real", ret_dtype=result_dtype)
+
+
+def imag(x: Array) -> ArrayOrScalar:
+    if x.dtype.kind == "c":
+        result_dtype = np.empty(0, dtype=x.dtype).real.dtype
+    else:
+        result_dtype = x.dtype
```
Comment (inducer) [inline pytato/array.py:1957]:
```suggestion
        return zeros_like(x)
```

File: pytato/array.py [hunk 3]
```diff
@@ -1926,6 +1941,22 @@ def log10(x: Array) -> ArrayOrScalar:
 def isnan(x: Array) -> ArrayOrScalar:
     return _apply_elem_wise_func((x,), "isnan", np.dtype(np.int32))
 
+
+def real(x: Array) -> ArrayOrScalar:
+    if x.dtype.kind == "c":
+        result_dtype = np.empty(0, dtype=x.dtype).real.dtype
+    else:
+        result_dtype = x.dtype
```
Comment (inducer) [inline pytato/array.py:1949]:
```suggestion
        return x
```

File: test/test_codegen.py [hunk 4]
```diff
@@ -533,7 +533,7 @@ def test_math_functions(ctx_factory, dtype, function_name):
                                                          "arctan", "log10"]:
         pytest.skip("Unsupported by loopy.")
 
-    if np.dtype(dtype).kind == "f" and function_name in ["conj"]:
+    if np.dtype(dtype).kind == "f" and function_name in ["conj", "real", "imag"]:
         pytest.skip("Unsupported by loopy.")
```
Comment (inducer) [inline test/test_codegen.py:538]:
```suggestion
```

**Review by inducer — COMMENTED** [review]
Thanks! I think `real`, `imag` and `conj` are actually no problem to implement for reals here.

File: pytato/array.py [hunk 5]
```diff
@@ -569,6 +574,16 @@ def __abs__(self) -> Array:
     def __pos__(self) -> Array:
         return self
 
+    @property
```
Comment (matthiasdiener) [inline pytato/array.py:577]: I'm not sure if this actually needs to be a `@proprty`

Comment (inducer) [reply pytato/array.py:577]: I think it does.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**